### PR TITLE
[AST] Block recursion through ProtocolDecl::getInheritedProtocols().

### PR DESCRIPTION
**Pull request by DougGregor**

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

File: include/swift/AST/Decl.h [hunk 1]
```diff
@@ -520,7 +520,10 @@ class alignas(1 << DeclAlignInBits) Decl {
     KnownProtocol : 8, // '8' for speed. This only needs 6.
 
     /// The number of requirements in the requirement signature.
-    NumRequirementsInSignature : 16
+		NumRequirementsInSignature : 16,
```
Comment (LuizZak) [inline include/swift/AST/Decl.h]: Seems like this, and other changes, are indented with tabs instead of spaces, like the rest of the file?

Comment (DougGregor) [reply include/swift/AST/Decl.h]: Yikes, good catch. My editor has gone nuts.

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

File: include/swift/AST/Decl.h (hunk 1 above)
Comment (DougGregor) [reply include/swift/AST/Decl.h]: (I'll clean them up shortly; for now, I need to unblock CI)

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

File: include/swift/AST/Decl.h (hunk 1 above)
Comment (DougGregor) [reply include/swift/AST/Decl.h]: Whitespace cleaned up in https://github.com/apple/swift/pull/17277, thank you!

**Pull request merged**